configs/template_config.yaml

```diff
@@ -44,12 +44,41 @@ models_path: "/YOUR/PATH/HERE/data/models"
 # TensorBoard Path
 tboard_path: ~
 
+# Checkpoint Path: path to checkpoint to load for generation/inference
+checkpoint_path: ~
+
 # Splits: Train split, Validation Split, and Test Split
 splits:
   - 0.7
   - 0.2
   - 0.1
 
+# Device Configuration
+device: "cuda" # Options: "cpu", "cuda"
+# num_nodes (int): Number of nodes
+num_nodes: 1
+# Number of GPUs (int)
+num_devices: 1 
+# strategy (string): Distributed strategy for training. You probably don't need to change this
+strategy: "ddp" # Options: "ddp", "ddp_spawn". 
+
+# Training Configuration
+rand_seed: 42
+# epochs (int): Number of epochs to train for.
+epochs: 1
+# lr (float): Learning rate of model to train.
+learning_rate: 0.001 # lr
+# save_top_k (int): Number of best models to save.
+save_top_k: 3
+# val_check_interval (float): How often to validate as a fraction of an epoch.
+val_check_interval: 0.5
+# use_slurm (bool): Whether to use SLURM for training.
+use_slurm: true
+# gamma (float): Learning rate scheduler gamma.
+gamma: 0.85
+# accumulate_grad_batches (int): Accumulate gradients over n batches.
+accumulate_grad_batches: 1
+
```

Comment:
Is there a reason that we're pulling this out of the "MODEL TRAINING CONFIGURATION" section below?

Reply:
I would argue yes. Model Training Configuration should have the parameters for actual model configuration, i.e., layers, width, etc. These parameters are settings for the devices and the hyperparameters, such as LR, epochs, etc. I would say separating them makes it easier to parse through. These also need to be reorganized a bit.